mirascope/core/base/prompt.py

```diff
@@ -372,15 +509,23 @@ def recommend_book(genre: str):
             attribute of the decorated input prompt or call.
     """
 
+    if template_or_message_function is None:
+        # For @prompt_template() case
+        return _messages_decorator()
+
+    if _is_message_param_function(template_or_message_function):
+        # For @prompt_template case
+        return _messages_decorator()(template_or_message_function)
+
```

Comment:
Two cases can be supported: `@prompt_template()` and `@prompt_template`.

Reply:
instead of refactoring and supporting both cases, should we just write the single case here where they write `@prompt_template()` and then type it to expect the `Messages` return type?